Resources/public/js/views/ez-universaldiscoveryview.js

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-universaldiscoveryview', function (Y) {
+    "use strict";
+    /**
+     * Provides the Universal discovery View class
+     *
+     * @module ez-universaldiscoveryview
+     */
+    Y.namespace('eZ');
+
+    var DISCOVERED = 'contentDiscovered',
+        CANCEL = 'cancelDiscover';
+
+    /**
+     * The universal discovery view is a widget to allow the user to pick one or
+     * several contents in the repository.
+     *
+     * @namespace eZ
+     * @class UniversalDiscoveryView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UniversalDiscoveryView = Y.Base.create('universalDiscoveryView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-universaldiscovery-confirm': {
+                'tap': '_confirmSelection'
+            },
+
+            '.ez-universaldiscovery-cancel': {
+                'tap': '_cancel',
+            },
+        },
+
+        initializer: function () {
+            this.on('contentDiscoveredHandlerChange', function (e) {
+                this._syncEventHandler(DISCOVERED, e.prevVal, e.newVal);
+            });
```

Comment:
nitpick: you could write it like this: `this.on('contentDiscoveredHandlerChange', this._syncEventHandler, this, DISCOVERED)` and make use of event param in the `this._syncEventHandler` method.